response_operations_ui/views/respondents.py

```diff
@@ -47,17 +47,29 @@ def search_redirect():
     form = RespondentSearchForm()
     form_valid = form.validate()
 
-    if not form_valid:
-        flash("At least one input should be filled")
-        return redirect(url_for("respondent_bp.respondent_home"))
-
-    email_address = form.email_address.data or ""
-    first_name = form.first_name.data or ""
-    last_name = form.last_name.data or ""
+    if request.method == "POST":
+        if not form_valid:
+            flash("At least one input should be filled")
+            return redirect(url_for("respondent_bp.respondent_home"))
+        email_address = form.email_address.data or ""
+        first_name = form.first_name.data or ""
+        last_name = form.last_name.data or ""
+    else:
+        email_address = request.args.get("email", "")
+        first_name = request.args.get("firstname", "")
+        last_name = request.args.get("lastname", "")
+    page = request.values.get("page", "1")
 
+    pagination_href = "search?"
```

Comment:
Can you factor this into it's own method somewhere in this file and write a noddy unit test for it?